build.sc

```diff
@@ -1559,191 +1650,301 @@ object dev extends MillModule {
+      case l if l.startsWith("version:") =>
+        s"version: 'master'" + "\n" + s"display-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-version:") =>
+        s"    mill-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-last-tag:") =>
+        s"    mill-last-tag: '${millLastTag()}'"
+      case l => l
     }
-    def generatePages(authorMode: Boolean) = T.task {
-      // dependency to sources
-      source()
-      val docSite = T.dest
-      val playbook = docSite / "antora-playbook.yml"
-      val siteDir = docSite / "site"
-      os.write(
-        target = playbook,
-        data = githubPagesPlaybookText(authorMode)(),
-        createFolders = true
-      )
-      // check xrefs
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          "--generator",
-          "@antora/xref-validator",
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        ) ++
-          Seq("--fetch").filter(_ => !authorMode)
-      )
-      // generate site (we can skip the --fetch now)
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        )
+    os.write.over(dest / "antora.yml", lines.mkString("\n"))
+    PathRef(dest)
+  }
+  def githubPagesPlaybookText(authorMode: Boolean): Task[String] = T.task {
+    s"""site:
+       |  title: Mill
+       |  url: ${if (authorMode) s"${T.dest}/site/index.html" else Settings.docUrl}
+       |  start_page: mill::Intro_to_Mill.adoc
+       |
+       |content:
+       |  sources:
+       |    - url: ${if (authorMode) baseDir else Settings.projectUrl}
+       |      branches: []
+       |      tags: ["0.10.12", "0.9.12"]
+       |      start_path: docs/antora
+       |    # the master documentation (always in author mode)
+       |    - url: ${baseDir}
+       |      # edit_url: ${Settings.projectUrl}/edit/{refname}/{path}
+       |      branches: HEAD
+       |      start_path: ${devAntoraSources().path.relativeTo(baseDir)}
+       |ui:
+       |  bundle:
+       |    url: https://gitlab.com/antora/antora-ui-default/-/jobs/artifacts/master/raw/build/ui-bundle.zip?job=bundle-stable
+       |    snapshot: true
+       |  supplemental_files: ${supplementalFiles().path.toString()}
+       |
+       |asciidoc:
+       |  attributes:
+       |    mill-github-url: ${Settings.projectUrl}
+       |    mill-doc-url: ${if (authorMode) s"file://${T.dest}/site" else Settings.docUrl}
+       |    mill-download-url: ${if (authorMode) s"file://${exampleZips().head.path / os.up}" else "https://github.com/com-lihaoyi/mill/releases/latest"}
+       |    mill-example-url: ${if (authorMode) s"file://${T.workspace}" else "https://github.com/com-lihaoyi/mill/blob/main/"}
```

Comment:
Tested these URLs to work for the local version of the doc-site. Hopefully they work when published

Reply:
We should reuse already defined variables where possible, e.g. `Settings.projectUrl` instead of `https://github.com/com-lihaoyi/mill` and others.